seis/ms/header.go

```diff
@@ -0,0 +1,359 @@
+package ms
+
+import (
+	"encoding/binary"
+	"fmt"
+	//TODO: needs v 1.15 "hash/maphash"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// RecordHeaderSize is the miniseed block fixed header length.
+const RecordHeaderSize = 48
+
+type RecordHeader struct {
+	SequenceNumber [6]byte // ASCII String representing a 6 digit number
+
+	DataQualityIndicator byte // ASCII: D, R, Q or M
+	ReservedByte         byte // ASCII: Space
+
+	// These are ascii strings
+	StationIdentifier  [5]byte // ASCII: Left justify and pad with spaces
+	LocationIdentifier [2]byte // ASCII: Left justify and pad with spaces
+	ChannelIdentifier  [3]byte // ASCII: Left justify and pad with spaces
+	NetworkIdentifier  [2]byte // ASCII: Left justify and pad with spaces
+
+	RecordStartTime      BTime  // Start time of record
+	NumberOfSamples      uint16 // Number of Samples in the data block which may or may not be unpacked.
+	SampleRateFactor     int16  // >0: Samples/Second <0: Second/Samples 0: Seconds/Sample, ASCII/OPAQUE DATA records
+	SampleRateMultiplier int16  // >0: Multiplication Factor <0: Division Factor
+
+	// Flags are bit masks
+	ActivityFlags    byte
+	IOAndClockFlags  byte
+	DataQualityFlags byte
+
+	NumberOfBlockettesThatFollow uint8 // Total number of blockettes that follow
+
+	TimeCorrection  int32  // 0.0001 second units
+	BeginningOfData uint16 // Offset in bytes to the beginning of data.
+	FirstBlockette  uint16 // Offset in bytes to the first data blockette in the data record.
+}
+
+func (h RecordHeader) SeqNumber() int {
+	s := string(h.SequenceNumber[:])
+	if no, err := strconv.ParseInt(s, 10, 32); err == nil {
+		return int(no)
+	}
+	return 0
+}
+func (h *RecordHeader) SetSeqNumber(no int) {
+	copy(h.SequenceNumber[:], []byte(fmt.Sprintf("%06d", no)))
+}
+
+func (h RecordHeader) Station() string {
+	return strings.TrimSpace(string(h.StationIdentifier[:]))
+}
+func (h *RecordHeader) SetStation(s string) {
+	for len(s) < 5 {
+		s += " "
+	}
+	copy(h.StationIdentifier[:], s)
```

Comment:
I think `s` needs to be casted to `[]byte`

Reply:
from https://golang.org/ref/spec#Appending_and_copying_slices

> As a special case, copy also accepts a destination argument assignable to type []byte with a source argument of a string type. This form copies the bytes from the string into the byte slice.